### PR TITLE
Add example to flutter_lints package

### PR DESCRIPTION
**Pull request by goderbauer**
for more points on pub.dev :) 

Inspired by https://github.com/dart-lang/lints/pull/36.

### CODE REVIEW

**Review by pq — APPROVED** [review]
Nice!

**Comment by pq** [comment]
@goderbauer or @devoncarew, if you want to lift the example source and drop it into `package:lints` I say go for it.

Note that if we add source, we'll need to update this validation though:

https://github.com/dart-lang/lints/blob/f9670df2a66e0ec12eb51554e70c1cbf56c8f5d0/.github/workflows/validate.yml#L24

**Comment by goderbauer** [comment]
> Note that if we add source, we'll need to update this validation though:

Oh, right. I will have to do the same for the check in this package...

File: packages/flutter_lints/run_tests.sh [hunk 1]
```diff
@@ -3,6 +3,7 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
+# Dart sources are only allowed in the example directory.
```
Comment (pq) [inline packages/flutter_lints/run_tests.sh:6]: 👍

File: packages/flutter_lints/example/README.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+packages, and plugins.
+
+The `flutter_lints` packages is listed as a dev_dependency in the `pubspec.yaml`
+file.
+
+The lint set provided by the package is activated in the `analysis_options.yaml`
+file. The lints enforced for this project can be further customized in that
+file.
+
+The dart code in this project (e.g. `lib/main.dart`) is analyzed using the
```
Comment (devoncarew) [inline packages/flutter_lints/example/README.md:12]: `dart` ==> `Dart`

**Review by devoncarew — APPROVED** [review]

**Pull request merged**